todo_app/app.py

```diff
@@ -1,44 +1,83 @@
+from todo_app.user.user import Roles, User
+import flask_login
+from todo_app.github.github_authentication import GithubAuthentication
 from todo_app.mongo.mongo_wrapper import MongoWrapper
 from flask import Flask, render_template, request, redirect, url_for
 import os
 from todo_app.models.view_model import ViewModel
+from flask_login import LoginManager, login_required
 
 
 def create_app():
     app = Flask(__name__)
+    app.config["SECRET_KEY"]=os.environ["SECRET_KEY"]
     mongo_database = MongoWrapper(os.environ['DEFAULT_DATABASE'])
+    login_manager = LoginManager()
+    github_authenticator = GithubAuthentication()
 
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        authentication_url = github_authenticator.get_github_identity()
```

Comment:
Small note on naming, it's surprising here that `get_github_identity` actually returns a URL - `get_github_auth_url` might be clearer